src/Configurable.h

```diff
@@ -161,6 +161,31 @@ class Configurable
             }
         }
     }
+
+    void config_hash(std::string &hash)
+    {
+        std::shared_lock lock(_config_mutex);
+        std::string data;
+        for (const auto &[key, value] : _config) {
+            std::visit([&data, key = key](auto &&arg) {
+                data += key;
+                using T = std::decay_t<decltype(arg)>;
+                if constexpr (std::is_same_v<T, StringList>) {
+                    for (const auto &s : arg) {
+                        data += s;
+                    }
+                } else if constexpr (std::is_same_v<T, std::string>) {
+                    data += arg;
+                } else {
+                    data += std::to_string(arg);
+                }
+            },
+                value);
+        }
+        std::sort(data.begin(), data.end());
```

Comment:
if this is just sorting all of the letters, i think there may be cases where the hash is not unique for some key/value combinations:

a=bc
b=ac

and 

b=bc
a=ac

both yield

aabbcc

even though the values for a and b are different.

If that's the case, I think we need to instead generate a string like this in `data`: "a=acb=bc" (sorted with key/val pairs intact) and hash that instead. Keeping the association solves the problem because we have "a=acb=bc" and "a=bcb=ac" which will hash differently.

Reply:
@weyrick So you think we can do a vector of strings where each string is "key+values" ? 
then sort the string vector and we concat the vector in one string and hash it

Reply:
yes!